openhsi/cameras.py

```diff
@@ -370,3 +372,35 @@ def set_exposure(self, exposure_ms:float):
         self.flircam.ExposureTime = self.settings["exposure_ms"]*1e3 # convert to us
 
 
+
+# Cell
+#hardware
+
+def switched_camera(
+    cam_name:str       = "FlirCamera", # Camera Class Name from openhsi.cameras
+    n_lines:int        = 128, # how many along-track pixels
+    processing_lvl:int = 0, # desired processing done in real time
+    json_path:str      = "/media/pi/fastssd/cals/flir_settings.json", # path to settings file
+    pkl_path:str       = "/media/pi/fastssd/cals/flir_calibration.pkl", # path to calibration file
+    preconfig_meta:str = "/media/pi/fastssd/cals/preconfig_metadata.json", # path to metadata file
+    ssd_dir:str        = "/media/pi/fastssd", # path to SSD
+    switch_pin:int     = 17, # button that controls collection
+):
+    """If `switch_pin` is high, collect with the camera until switch is low."""
+
+    try: import RPi.GPIO as GPIO
+    except ModuleNotFoundError:
+        try: import Jetson.GPIO as GPIO
+        except ModuleNotFoundError:
+            print("No module named RPi or Jetson. Did you try `pip install RPi.GPIO` or `pip install Jetson.GPIO`?")
+    GPIO.setmode(GPIO.BCM) # BCM pin-numbering scheme from Raspberry Pi
+    GPIO.setup(switch_pin, GPIO.IN)
+
+    if "Shared" in cam_name: eval(f"exec('from openhsi.shared import {cam_name}')")
+    else: eval(f"exec('from openhsi.cameras import {cam_name}')")
+    cam = FlirCamera(n_lines = n_lines, processing_lvl = processing_lvl,
+                     json_path = json_path,
+                     pkl_path  = pkl_path)
+    while GPIO.input(switch_pin) == True: # collect while go button is on.
+        cam.collect()
+        cam.save(ssd_dir,preconfig_meta_path=preconfig_meta)
```

Comment:
IMO this should be a seperate library, i.e. repo of your raspi deployment code, or part of the "real time script", not in OpenHSI library.

Reply:
I have similar feelings since it requires/assumes users are using a separate ancillary sensor board with it's own C++ code and this provides the interface for it.
But it is also a necessary part for the georectification and we wanted a "complete" hsi package. The majority is in the `sensors` module so most people won't use it by accident.
The reason why this lone function is here instead of the `sensors` module is because the Multiprocessing library requires the function it calls to be in a separate file.

Reply:
> I have similar feelings since it requires/assumes users are using a separate ancillary sensor board with it's own C++ code and this provides the interface for it.
> But it is also a necessary part for the georectification and we wanted a "complete" hsi package. The majority is in the `sensors` module so most people won't use it by accident.
> The reason why this lone function is here instead of the `sensors` module is because the Multiprocessing library requires the function it calls to be in a separate file.


I do see what you mean, but this a very specific implementation of how to collect the sensor data right? Rather than something generic. It's inclusion in an example might fit the current architecture better?

Reply:
the implementation is tied down to the specific format the sensor data packets are sent in. Could be made more generic but the specialisation is what makes it useful as it solves many of the problems that come with collecting the data in realtime. It is also very difficult to develop a similar capacity then integrate it with the openhsi library for simultaneous collect - this already took me a long time to develop. 
I plan to upload my C++ code, and fixed PCB files to the openhsi repository so it works together. Kind of like how the 3D models will be in a separate repo. I think that's ok. In replicating the lightweight system, the sensor code will be used again.